src/data/fintype/order.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2021 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+
+import order.lattice
+import data.fintype.basic
+
+/-!
+# Order structures on finite sets
+
+This file provides instances showing that a `semilattice_sup_bot` on a `fintype` gives a
+`complete_lattice`, and that a `linear_order` on a nonempty `fintype` gives a
+`complete_linear_order`. The latter applies in particular to `fin (n+1)`. The instances are
+noncomputable because the definition of `Sup` uses `set.to_finset`, which implicitly
+requires a `decidable_pred` instance for every `s : set α`.
+-/
+
+open_locale classical
+
+variables {α : Type*}
+
+@[priority 100] noncomputable instance fintype.complete_semilattice_Sup
```

Comment:
```suggestion
noncomputable def fintype.complete_semilattice_Sup
```
And similarly for the others, they all cause loops in type class search (because it goes from `linear_order α` to `complete_linear_order α` and then to `linear_order α` back again):
```lean
example : semilattice_sup_bot α := by apply_instance
/-
maximum class-instance resolution depth has been reached (the limit can be increased by setting option 'class.instance_max_depth') (the class-instance resolution trace can be visualized by setting option 'trace.class_instances')
state:
α : Type u_1
⊢ semilattice_sup_bot α
-/
```